packages/expo-doctor/src/checks/AppConfigFieldsNotSyncedToNativeProjectsCheck.ts

```diff
@@ -0,0 +1,77 @@
+import fs from 'fs';
+import path from 'path';
+
+import { DoctorCheck, DoctorCheckParams, DoctorCheckResult } from './checks.types';
+import { learnMore } from '../utils/TerminalLink';
+import { isFileIgnoredAsync } from '../utils/isFileIgnoredAsync';
+
+const appConfigFieldsToSyncWithNative = [
+  'ios',
+  'android',
+  'plugins',
+  'icon',
+  'scheme',
+  'userInterfaceStyle',
+  'splash',
+  'updates',
+  'orientation',
+  'backgroundColor',
+  'primaryColor',
+  'notification',
+  'androidStatusBar',
+  'androidNavigationBar',
+  'jsEngine',
+  'locales',
+];
+
+export class AppConfigFieldsNotSyncedToNativeProjectsCheck implements DoctorCheck {
+  description = 'Check for app config fields that may not be synced in a non-CNG project';
+
+  sdkVersionRange = '*';
+
+  async runAsync({
+    exp,
+    projectRoot,
+    staticConfigPath,
+    dynamicConfigPath,
+  }: DoctorCheckParams): Promise<DoctorCheckResult> {
+    const issues: string[] = [];
+    const appJsonFields: string[] = Object.keys(exp);
+    const unsyncedFields: string[] = [];
+    let advice;
+
+    // iterate over all fields in app.json and add those that will not be synced to unsyncedFields array
+    for (const field of appJsonFields) {
+      if (appConfigFieldsToSyncWithNative.includes(field)) {
+        unsyncedFields.push(field);
+      }
+    }
+
+    if (
+      unsyncedFields.length &&
+      // git check-ignore needs a specific file to check gitignore, we choose Podfile or build.gradle
+      ((await existsAndIsNotIgnoredAsync(path.join(projectRoot, 'ios', 'Podfile'))) ||
+        (await existsAndIsNotIgnoredAsync(path.join(projectRoot, 'android', 'build.gradle'))))
+    ) {
+      // get the name of the config file
+      const configFilePath = dynamicConfigPath ?? staticConfigPath;
+      const configFileName = path.basename(configFilePath ?? 'app.json');
+
+      issues.push(
+        `This project contains native project folders but also has native configuration properties in ${configFileName}, indicating it is configured to use Continuous Native Generation (CNG). When the 'android' / 'ios' folders are present, EAS Build will not sync the following properties: ${unsyncedFields.join(', ')}. \n`
```

Comment:
```suggestion
        `This project contains native project folders but also has native configuration properties in ${configFileName}, indicating it is configured to use Continuous Native Generation (CNG). When the android/ios folders are present, EAS Build will not sync the following properties: ${unsyncedFields.join(', ')}. \n`
```
I looked elsewhere, and this is how we referred to something similar in another check. The quotes make more sense when it's something you copy (like the gitignore snippets) than here. Apologies for the thrash.